Source/RapyutaSimulationPlugins/Public/Robots/RRRobotStructs.h

```diff
@@ -1119,31 +1119,76 @@ struct FRRRobotModelData
                                 { return (InWheelProp.WheelName == InLinkName); });
     }
 
-    const FRRRobotLinkProperty FindLinkProp(int8 InLinkIndex) const
+    FRRRobotLinkProperty GetBaseLinkProp() const
     {
         for (const auto& linkProp : LinkPropList)
         {
-            if (linkProp.LinkIndex == InLinkIndex)
+            if (linkProp.Name == BaseLinkName)
             {
                 return linkProp;
             }
         }
+        return FRRRobotLinkProperty();
+    }
 
+    FRRRobotLinkProperty GetLinkProp(const FString& InLinkName) const
+    {
+        for (const auto& linkProp : LinkPropList)
+        {
+            if (linkProp.Name == InLinkName)
+            {
+                return linkProp;
+            }
+        }
         return FRRRobotLinkProperty();
     }
 
-    const FRRRobotJointProperty FindJointPropByLinkIndex(int8 InLinkIndex) const
+    FRRRobotJointProperty GetJointProp(const FString& InJointName) const
     {
-        const FRRRobotLinkProperty linkProp = FindLinkProp(InLinkIndex);
+        for (const auto& jointProp : JointPropList)
+        {
+            if (jointProp.Name == InJointName)
+            {
+                return jointProp;
+            }
+        }
+        return FRRRobotJointProperty();
+    }
 
+    FRRRobotJointProperty GetJointPropFromChildLinkName(const FString& InChildLinkName) const
```

Comment:
it is better to add comment that this method return first joint hose childlink name is InChildLinkName.
We may support parallelink robot in the future which have multiple links connects to same link.